frontend/src/scenes/heatmaps/HeatmapsBrowser.tsx

```diff
@@ -211,26 +234,28 @@ function EmbeddedHeatmapBrowser({
     return browserUrl ? (
         <div className="flex flex-row gap-x-2 w-full">
             <FilterPanel />
-            <iframe
-                ref={iframeRef}
-                className="flex-1"
-                src={appEditorUrl(browserUrl, {
-                    userIntent: 'heatmaps',
-                })}
-                // eslint-disable-next-line react/forbid-dom-props
-                style={{
-                    background: '#FFF',
-                }}
-                onLoad={onIframeLoad}
-                // these two sandbox values are necessary so that the site and toolbar can run
-                // this is a very loose sandbox,
-                // but we specify it so that at least other capabilities are denied
-                sandbox="allow-scripts allow-same-origin"
-                // we don't allow things such as camera access though
-                allow=""
-            />
-
-            {loading && <SpinnerOverlay />}
+            <div className="relative flex-1 w-full h-full">
+                {iframeBanner && <IframeErrorOverlay />}
+                <LoadingOverlay />
```

Comment:
It's a personal rule but I think it's always better to conditionally render a component outside the component:
```suggestion
                {loading && <LoadingOverlay />}
```

Reply:
ELI5 why you prefer that? being explicit?

(not challenging just for clarity 😊)

Reply:
<img width="456" alt="Screenshot 2024-06-10 at 09 45 16" src="https://github.com/PostHog/posthog/assets/984817/ca2ea81d-56f1-462b-9917-0d6325c828e5">

the funny thing was `IFrameErrorOverlay` already internally managed its conditional display so could be simplified like this

Reply:
1. Mainly consistency I guess. The neighbouring component is very similar but is conditional outside the component. I might find myself assuming that the `LoadingOverlay` is always rendered if I based it off how I see the `IframeErrorOverlay` works. 
2. Also makes the component stateless (won't always be true but is the case here). In theory should make it more reusable
3. It's also easier to reason about the size of the component if it is always rendered. Otherwise you have to remember as an implementor that the component can sometimes render nothing and account for that in your wrapping layout. Not going to miss that if the null case has to be accounted for in your implementation.
4. Testing is easier because you can pass values in

Reply:
Ah, I missed that 😄 I would say either works (but go with one approach in this component given the proximity) because I don't think we will achieve / should optimise for this kind of consistency across the codebase

Reply:
i'll let this seep into my subconscious... i've given up hoping that React will die so I guess I should develop some opinions 🤣